flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
+            merge=True,
         )
 
-    def add_wind_forcing_from_grid(self, ds: xr.DataArray):
+    ### PROJECTIONS HELPERS - not needed at the moment ###
+
+    ##### OLD CODE BELOW #####
+    # @Gundula: I have renamed some the functions and added comments to some that explain when they were called in the original hazard.py just to keep track.
+    # Im not super familiar with hydromt_sfincs and its functions, so I have not changed the logic of the functions, only the names and comments.
+    # It could be the case that some of the functions in here are not needed anymore, but we can always remove them later if needed.
+
+    # I have also added a '_' to the start of all functions I see as private.
+    # Public functions can be called from outside of the class, all private functions should not be called from outside of the class.
+    # (even though python does not care and will allow it anyways, we can still follow the convention just to make it clear)
+
+    ### SFINCS SETTERS ###
+    def _set_timing(self, event: EventModel):
+        """Set model reference times based on event time series."""
+        # Get start and end time of event
+        tstart = event.time.start_time
+        tstop = event.time.end_time
+
+        # Update timing of the model
+        self._model.set_config("tref", tstart)
+        self._model.set_config("tstart", tstart)
+        self._model.set_config("tstop", tstop)
+
+    def _set_config_spw(self, spw_name: str):
+        self._model.set_config("spwfile", spw_name)
+
+    def _turn_off_bnd_press_correction(self):
+        self._model.set_config("pavbnd", -9999)
+
+    ### ADD TO SFINCS ###
+    # @Gundula functions starting with `add` should not blindly overwrite data in sfincs, but read data from the model, add something to it, then set the model again.
+    # This is to ensure that we do not overwrite any existing data in the model.
+    # (maybe this is what hydromt_sfincs already does, but just checking with you)
+    def _add_obs_points(self):
+        """Add observation points provided in the site toml to SFINCS model."""
+        if self._site.attrs.obs_point is not None:
+            logging.info("Adding observation points to the overland flood model...")
+
+            obs_points = self._site.attrs.obs_point
+            names = []
+            lat = []
+            lon = []
+            for pt in obs_points:
+                names.append(pt.name)
+                lat.append(pt.lat)
+                lon.append(pt.lon)
+
+            # create GeoDataFrame from obs_points in site file
+            df = pd.DataFrame({"name": names})
+            gdf = gpd.GeoDataFrame(
+                df, geometry=gpd.points_from_xy(lon, lat), crs="EPSG:4326"
+            )
+
+            # Add locations to SFINCS file
+            self._model.setup_observation_points(locations=gdf, merge=False)
+
+    def _add_wind_forcing_from_grid(self, ds: xr.DataArray):
```

Comment:
that should go to add_forcing -> wind if wind from hindcast model
it should also be called for offshore model for historic_offshore events (i.e. water level forcing from model (oh, that makes it a little messier again)

Reply:
I dont think that is messy if it is just adding this function call to the add_forcing_x functions where its needed? or am i not understanding you correctly?